crates/starknet-server/src/main.rs

```diff
@@ -57,7 +64,22 @@ async fn main() -> Result<(), anyhow::Error> {
         IpAddr::from_str(starknet_config.host.as_str()).expect("Invalid value for host IP address");
     let mut addr = SocketAddr::new(host, starknet_config.port);
 
-    let api = api::Api::new(Starknet::new(&starknet_config)?);
+    // Load starknet transactions
+    let mut transactions: StarknetTransactions = StarknetTransactions::default();
+    if let Some(path) = &starknet_config.dump_path {
+        let file_path = Path::new(path);
+        if file_path.exists() {
+            let mut file = File::open(file_path).expect("Failed to open file");
+            let mut v: Vec<u8> = Vec::new();
+            file.read_to_end(&mut v).expect("Failed to read from file");
+            let decoded: Option<String> =
+                bincode::deserialize(&v[..]).expect("Failed to deserialize starknet transactions");
+            transactions = serde_json::from_str(decoded.unwrap().as_str())
+                .expect("Failed to decode starknet transactions");
+        }
+    }
+
+    let api = api::Api::new(Starknet::new(&starknet_config, Some(transactions))?);
```

Comment:
Isn't that always some even though `transactions` can be empty?

Reply:
Fixed in 3028e91e1f57b71f5d9daae23a39ce7bfbce8b90